packages/camera/camera_android_camerax/lib/src/android_camera_camerax.dart

```diff
@@ -981,4 +1062,89 @@ class AndroidCameraCameraX extends CameraPlatform {
+
+  /// Starts a foucs and metering action.
+  ///
+  /// This method will merge the current action's metering points with that
+  /// specified such that only one auto-exposure metering point and one
+  /// auto-focus metering point will be part of the action. This method would
+  /// also accommodate at most one auto-white balance metering point, but that
+  /// mode is currently unused in this plugin.
+  ///
+  /// Thus, if [meteringPoint] is non-null, this action includes:
+  ///   * metering points and their modes previously added to
+  ///     [currentFocusMeteringAction] that do not share a metering mode with
+  ///     [meteringPoint] and
+  ///   * [meteringPoint] with the specified [meteringMode].
+  /// If [meteringPoint] is null, this action includes only metering points and
+  /// their modes previously added to [currentFocusMeteringAction] that do not
+  /// share a metering mode with [meteringPoint]. If there are no such metering
+  /// points, then the previously enabled focus and metering actions will be
+  /// canceled.
+  Future<void> _startFocusAndMeteringFor(
+      {required Point<double>? meteringPoint,
+      required int meteringMode}) async {
+    if (meteringPoint == null) {
+      // Try to clear any metering point from previous action with the specified
+      // meteringMode.
+      if (currentFocusMeteringAction == null) {
+        // Attempting to clear a metering point from a previous action, but no
+        // such action exists.
+        return;
+      }
+
+      // Remove metering point with specified meteringMode from current focus
+      // and metering action, as only one focus or exposure point may be set
+      // at once in this plugin.
+      final List<(MeteringPoint, int?)> newMeteringPointInfos =
+          currentFocusMeteringAction!.meteringPointInfos
+              .where(((MeteringPoint, int?) meteringPointInfo) =>
+                  // meteringPointInfo may technically include points without a
+                  // mode specified, but this logic is safe because this plugin
+                  // only uses points that explicitly have mode
+                  // FocusMeteringAction.flagAe or FocusMeteringAction.flagAf.
+                  meteringPointInfo.$2 != meteringMode)
+              .toList();
+
+      if (newMeteringPointInfos.isEmpty) {
+        // If no other metering points were specified, cancel any previously
+        // started focus and metering actions.
+        await cameraControl!.cancelFocusAndMetering();
+        return;
+      }
+      currentFocusMeteringAction =
+          proxy.createFocusMeteringAction(newMeteringPointInfos);
+    } else if (meteringPoint.x < 0 ||
+        meteringPoint.x > 1 ||
+        meteringPoint.y < 0 && meteringPoint.y > 1) {
+      throw CameraException('meteringPointInvalid',
+          'The coordinates of a metering point for an auto-focus or auto-exposure action must be within (0,0) and (1,1).');
```

Comment:
Nit: include the provided (actual) coordinates in the log message, something like 
> ...must be within (0,0) and (1,1), but point $meteringPoint was provided for metering mode $meteringMode